XHR.cgi

```diff
@@ -0,0 +1,15 @@
+#!/usr/local/bin/perl
+# A caller for loading XHR related routines
+use strict;
+
+our ($root_directory);
+
+BEGIN { push(@INC, "."); };
+use WebminCore;
+
+&init_config();
+&ReadParse();
+&switch_to_remote_user();
+do "$root_directory/XHR-lib.pl";
```

Comment:
Shouldn't there be some kind of access control on this CGI, so that it can only be called by users with access to the `mailbox` or `mailboxes` modules? Otherwise any Virtualmin user could potentially use (or abuse) it to fetch URLs through the server, which is a potential DOS attack risk.

Seems like we could move it into `mailboxes` , and also remove XHR-lib.pl and put the one `xhr` function into XHR.cgi 

Also, why is it XHR.cgi instead of xhr.cgi to be consistent with the rest of Webmin?

Reply:
> Also, why is it XHR.cgi instead of xhr.cgi to be consistent with the rest of Webmin?

It conflicts with the Authentic theme, which is already using `xhr.cgi`.

> Shouldn't there be some kind of access control on this CGI, so that it can only be called by users with access to the mailbox or mailboxes modules? 

I thought `&init_config();` would take care of it?

> Seems like we could move it into mailboxes , and also remove XHR-lib.pl and put the one xhr function into XHR.cgi

I was hoping to use it universally for other modules too..

Reply:
So we have both XHR.cgi and xhr.cgi ? That's a little confusing ... I'd suggest renaming one.

Also, init_config in a top-level CGI script won't check for any module access, because it's outside of modules.